learn/advanced/storage.md

```diff
@@ -71,3 +71,9 @@ These metrics are highly dependent on the machine that is running Meilisearch. R
 It is important to note that **there is no reliable way to predict the final size of a database**. This is true for just about any search engine on the market—we're just the only ones saying it out loud.
 
 Database size is affected by a large number of criteria, including settings, relevancy rules, use of facets, the number of different languages present, and more.
+
+## Soft deletion
+
+Meilisearch renders deleted documents inaccessible to all users but does not immediately remove them from the database. This is a common optimization technique called "soft deletion". Soft deleted documents are permanently deleted at a later date—the precise time depends on index size and available disk space. It might be important to check how soft deletion interacts with relevant data retention legislation.
```

Comment:
```suggestion
Meilisearch renders deleted documents inaccessible to all users but does not immediately remove them from the database. This is a common optimization technique called "soft deletion". Soft deleted documents are permanently deleted during a later update—the precise time depends on index size and available disk space. It might be important to check how soft deletion interacts with relevant data retention legislation.
```

Meilisearch won't delete your documents depending on how long they stayed in the database, the available processing power, or anything else.
The only time it checks if it needs to delete your soft-deleted documents is at the end of an update. So maybe this slight change in wording would avoid us some questions?

Reply:
Ah, wow, I did not know that at all. Your wording is definitely clearer in that respect, thanks—I'm making a couple of minor edits to your suggestion for better text flow and accepting it.